core/domain/classifier_domain.py

```diff
@@ -72,11 +116,52 @@ def to_dict(self):
         """
 
         return {
-            'classifier_id': self.id,
-            'exp_id': self.exp_id,
-            'exp_version_when_created': self.exp_version_when_created,
-            'state_name': self.state_name,
-            'algorithm_id': self.algorithm_id,
-            'cached_classifier_data': self.cached_classifier_data,
-            'data_schema_version': self.data_schema_version
+            'classifier_id': self._id,
+            'exp_id': self._exp_id,
+            'exp_version_when_created': self._exp_version_when_created,
+            'state_name': self._state_name,
+            'algorithm_id': self._algorithm_id,
+            'cached_classifier_data': self._cached_classifier_data,
+            'data_schema_version': self._data_schema_version
         }
+
+    def validate(self):
+        """Validates the classifier before it is saved to storage."""
+
+        if not isinstance(self.id, basestring):
+            raise utils.ValidationError(
+                'Expected id to be a string, received %s' % self.id)
+
+        if not isinstance(self.exp_id, basestring):
+            raise utils.ValidationError(
+                'Expected exp_id to be a string, received %s' % self.exp_id)
+        utils.require_valid_name(self.exp_id, 'the exploration id')
```

Comment:
Why does the exploration id have to be a "valid name"? Doesn't seem necessary to me since it's not human readable.